l1-contracts/src/core/libraries/rollup/RewardLib.sol

```diff
@@ -89,14 +89,17 @@ library RewardLib {
 
       // We can use fancier bitmaps for performance
       require(
-        !rewardStorage.proverClaimed[msg.sender][_epochs[i]],
+        !rewardStorage.proverClaimed[msg.sender].get(Epoch.unwrap(_epochs[i])),
         Errors.Rollup__AlreadyClaimed(msg.sender, _epochs[i])
       );
-      rewardStorage.proverClaimed[msg.sender][_epochs[i]] = true;
+      rewardStorage.proverClaimed[msg.sender].set(Epoch.unwrap(_epochs[i]));
 
       EpochRewards storage e = rewardStorage.epochRewards[_epochs[i]];
-      if (e.subEpoch[e.longestProvenLength].hasSubmitted[msg.sender]) {
-        accumulatedRewards += (e.rewards / e.subEpoch[e.longestProvenLength].summedCount);
+      SubEpochRewards storage se = e.subEpoch[e.longestProvenLength];
+      uint256 shares = se.shares[msg.sender];
+      if (shares > 0) {
```

Comment:
We keep this if to skip some reads in case we don't have any shares for the epoch.